### ISSUE TITLE
remove SSH services in all blockchain solutions

### ISSUE THREAD

**Issue by xmonader** [threefoldtech/vdc-solutions-charts#49]
![image](https://user-images.githubusercontent.com/64129/104710645-ff643e00-5728-11eb-994e-35228e3ebb97.png)
- [x] dash
- [x] digibyte 
- [ ] harmony
- [ ] tomochain

**Comment by abdulgig** [comment]
Terminal access is a requirement here. Please dont expect  users to install an additional kubernetes client and manage their nodes. This has already been discussed

**Comment by xmonader** [comment]
You don't deploy a service and manually ssh into the pod and change its configurations. this is not how kubernetes works, if anything provide a better deployment with the right configurations from the beginning if you want to update you don't ssh and do that manually this completely defeats the purpose of Kubernetes!!

Also, if you need terminal access to do X use kubectl exec `which shouldn't be for anything except for debugging even`

we won't be allowing any ssh access without a compelling reason.

**Comment by abdulgig** [comment]
Have you gone through their official documentation ? Please do !  Perhaps it can help you find an optimum balance between philosophy and practicality. None of these partners have nodes or validators running on Kubernetes. We would be the first ones doing it actually, which is good, but few things to consider please,

1. To setup validators, you need to import wallets and enter private keys.  Users won't be entering their private keys to our chat flows on the marketplace. Under blockchain nodes, few things cannot be baked into ENV variables unless every user runs his private image or chart. They will have to be done manually, simple !

2. Using ```ssh``` or ```kubectl``` is really trivial as both can do the job. The only issue with kubectl is that not necessarily all users may be familiar with it and it may be cumbersome for them to manage their nodes. I had raised the same point during the product call this week

**Comment by xmonader** [comment]
if we are the first to do in Kubernetes then we do it "correctly". if you want to deploy in Kubernetes you do it the kubernetes way. proper images, proper deployments, proper configs, proper secret management, proper volumes and proper charts.  


>  you need to import wallets and enter private keys. Users won't be entering their private keys to our chat flows on the marketplace

- You understand that chatflows of the marketplace are local to the user "run in his own 3bot" not on the central VDC deployer right?
- You understand we even have full access to the deployed cluster too?

> The only issue with kubectl is that not necessarily all users may be familiar with it and it may be cumbersome for them to manage their nodes.

if their nodes run on kubernetes they use kubernetes tooling to manage it. - and certainly, even kubectl exec not even an option- 

to sum it up: `there's no point of using kubernetes if you will keep updating the deployments by sshing into the pods ` and if that's the case better off using some VM somewhere or a container somewhere "OUTSIDE" of kubernetes.

**Comment by sasha-astiadi** [comment]
@abdulgig @ashraffouda if its possible let us have a call on sunday to clear things up

**Comment by abdulgig** [comment]
Looks like pointless to debate here,

@xmonader In that case, please add the necessary instruction for kubectl in the chatflow https://github.com/threefoldtech/js-sdk/issues/2209

**Comment by xmonader** [comment]
> Looks like pointless to debate here,
> 
> @xmonader In that case, please add the necessary instruction for kubectl in the chatflow [threefoldtech/js-sdk#2209](https://github.com/threefoldtech/js-sdk/issues/2209)

Actually, we _shouldn't_ even do any terminal operations against the pod with `kubectl exec` nor _instruct_ the user to execute them. Proper configurations, secret management and proper container entrypoint need to be implemented on all of the blockchain solutions.

@ashraffouda @OmarElawady please make sure there's an issue with details needed to be executed on these solutions to be accepted as solutions in the VDC marketplace

**Comment by robvanmieghem** [comment]
The thing is that the normal blockchain projects documentation probably describe an interactive process

The concept in containerized applications is that you don’t do that so the dockerfiles and helm charts should be adjusted not to require interactive sessions

but if needed add a script in the container at startup that reads/ writes to a kubernetes secret or something

but not go ssh-ing into the thing to configure it

**Comment by abdulgig** [comment]
@robvanmieghem  OK will figure out with you then

**Comment by abdulgig** [comment]
@robvanmieghem , as per our discussion, we already have the required startup scripts that run the full node with nothing to be done via the terminal. So that is already automated !

To setup validators, you need to import wallets and enter private keys. Users won't be entering their private keys to our chatflows over the web/online market place. We already had a consensus with @despiegk on this !

Unlike other solutions which are web driven here, blockchain nodes are managed over the terminal by the users in the community, primarily dgb/dash wallets. Moreover, the user experience has to be consistent or better than what they already get.

I will share a spec with you and @despiegk so there is a clear understanding of what needs to be done

**Comment by robvanmieghem** [comment]
Exactly the reason why upfront specs are so important, if not, everyone has their own opinion and does not know the reasoning, nor can someone contribute ideas in time.

**Comment by abdulgig** [comment]
This has been settled and we are moving forward on this one 

PR https://github.com/threefoldtech/vdc-solutions-charts/pull/59

**Comment by sasha-astiadi** [comment]
i know this has been settled, but @abdulgig please do share your spec so we all understand what you'd suggest is best for our partners, and we can discuss, adjust and confirm that this is how we'd wanna move forward for the next deployments in the future, and have kds review it too

**Comment by sasha-astiadi** [comment]
this story is obsolete. we first focus on digibyte, and we create issue per helm chart, not all.| for 2.4; digibyte helm chart #66
